ibm/resource_ibm_cos_bucket.go

```diff
@@ -532,6 +548,43 @@ func resourceIBMCOSBucketUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	//update the object versioning (object versioning)
+	if d.HasChange("versioning") {
+		var versioning, versioning_ok = d.GetOk("versioning")
+		if versioning_ok {
```

Comment:
You need to handle if user removes the block from TF then versioning shd be set to suspended